onadata/apps/api/tests/viewsets/test_data_viewset.py

```diff
@@ -513,26 +513,31 @@ def test_delete_submission(self):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
         count = self.xform.instances.all().count()
         self.assertEqual(before_count - 1, count)
+
+    def test_cannot_delete_submission_as_granted_user_but_not_service_account(self):
+        self._make_submissions()
+        before_count = self.xform.instances.all().count()
+        view = DataViewSet.as_view({'delete': 'destroy'})
+        request = self.factory.delete('/', **self.extra)
+        formid = self.xform.pk
+
         self._create_user_and_login(username='alice', password='alice')
         # Allow Alice to delete submissions.
         assign_perm(CAN_VIEW_XFORM, self.user, self.xform)
         assign_perm(CAN_CHANGE_XFORM, self.user, self.xform)
-        self.extra = {
-            'HTTP_AUTHORIZATION': 'Token %s' % self.user.auth_token}
+        self.extra = {'HTTP_AUTHORIZATION': f'Token {self.user.auth_token}'}
         request = self.factory.delete('/', **self.extra)
         dataid = self.xform.instances.all().order_by('id')[0].pk
         response = view(request, pk=formid, dataid=dataid)
 
         # Alice cannot delete submissions with `CAN_CHANGE_XFORM`
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
-        # Now, she should be able to
+        # Even with correct permissions, Alice should not be able to delete
         remove_perm(CAN_CHANGE_XFORM, self.user, self.xform)
         assign_perm(CAN_DELETE_DATA_XFORM, self.user, self.xform)
         response = view(request, pk=formid, dataid=dataid)
-        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
-        count = self.xform.instances.all().count()
-        self.assertEqual(before_count - 2, count)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
I wonder if we should return a [405](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405) (or [501](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501)?) so that people who are still using this API (tsk, tsk) at least get some clue that no amount of wrangling with usernames, passwords, authentication tokens, or permissions will allow them to do this anymore